tensorflow/contrib/image/python/ops/image_ops.py

```diff
@@ -81,6 +81,48 @@ def rotate(images, angles, interpolation="NEAREST"):
     return output
 
 
+def translate(images, translations, interpolation="NEAREST"):
+  """Translate image(s) by the passed vectors(s).
+
+  Args:
+    images: A tensor of shape (num_images, num_rows, num_columns, num_channels)
+       (NHWC), (num_rows, num_columns, num_channels) (HWC), or
+       (num_rows, num_columns) (HW).
+    translations: A vector representing [dx, dy] or (if images has rank 4)
+        a vector of length num_images, with a [dx, dy] vector for each image in the batch.
+    interpolation: Interpolation mode. Supported values: "NEAREST", "BILINEAR".
+
+  Returns:
+    Image(s) with the same type and shape as `images`, translated by the given
+    vector(s). Empty space due to the translation will be filled with zeros.
+
+  Raises:
+    TypeError: If `image` is an invalid type.
+  """
+  image_or_images = ops.convert_to_tensor(images, name="images")
+  if image_or_images.dtype.base_dtype not in _IMAGE_DTYPES:
+    raise TypeError("Invalid dtype %s." % image_or_images.dtype)
+  if len(image_or_images.get_shape()) == 2:
+    images = image_or_images[None, :, :, None]
+  elif len(image_or_images.get_shape()) == 3:
+    images = image_or_images[None, :, :, :]
+  elif len(image_or_images.get_shape()) == 4:
+    images = image_or_images
+  else:
+    raise TypeError("Images should have rank between 2 and 4.")
+
+  output = transform(
```

Comment:
You should be able to remove everything but this statement because `transform` does the same checks, and we don't do anything else with `images` like we do in `rotate`.